### ISSUE TITLE
Cannot register time_t and size_t methods

### ISSUE THREAD

**Issue by marcizhu** [Aloshi/dukglue#19]
Hi, I've been using this library for a while (it's really great btw, good job!) but when I try to register a function from a class that returns a `time_t` or other types, I get some compilation errors.

Here is a minimal example to illustrate my point:
```cpp
#include <ctime>

class Foo
{
public:
    time_t getTime() const { return time(nullptr); }
};

// Somewhere else...
dukglue_register_method(ctx, &Foo::getTime, "getTime"); // This generates compilation errors
```

This 'bug' also happens with properties, and it's not a big deal since I can return an `int32_t` or even an `int64_t` if it was necessary, but it would be neat to use the native types.

Any help would be appreciated :)

**Comment by marcizhu** [comment]
Okay, that was simple. I fixed this with just one line of code (check it out [here](https://github.com/marcizhu/dukglue/commit/9ad8c07d231aef24f875d84c62c32457c675c499)). Should I make a PR?

**Comment by marcizhu** [comment]
I've investigated a bit and I realized this is a macOS-only bug: for some reason, clang on macOS treats `size_t` and `time_t` as different data types, but in Linux they are a simple typedef and generate a redefinition error.

I still have to investigate what happens on Windows, but simply adding an `#ifdef __APPLE__` around the line I added seems to fix the compilation errors on macOS and redefinition errors on Linux systems (using both clang and g++)

**Comment by Aloshi** [comment]
Nice catch, thanks for investigating. If you open a PR I'd be happy to merge.

**Comment by marcizhu** [comment]
Okay, I'll check if this fix also works on Windows and if so, in the next few hours I'll make a PR.

**Comment by marcizhu** [comment]
So in the last few days I've been testing more platforms and I've discovered a very similar thing happens on Raspberry Pi, which is an ARM platform and uses different data sizes for `time_t` and `size_t`. For some odd reason, `size_t` works but `time_t` doesn't. I've added a new commit to my pull request to fix this too (see commit 9a19177 on #20).

I realize this might be a bit "out of topic" since it is no longer macOS-only. If so, I can remove the last changes from my pull request (leaving only the fix for macOS) and open a new one with the fix for ARM. Just let me know and I'll do it in no time :)
